src/Toml/Bi/Map.hs

```diff
@@ -381,6 +383,29 @@ _BoundedInteger = BiMap (Right . toInteger) eitherBounded
          in Left $ ArbitraryError msg
       | otherwise = Right (fromIntegral n)
 
+-- | Helper bimap for 'SumType' and 'Data.Text.Text'.
+_SumTypeText :: (Read a, Show a, Enum a, Bounded a) => TomlBiMap a Text
+_SumTypeText = BiMap
+    { forward  = Right . tShow
+    , backward = toSumType
+    }
+  where
+    toSumType :: forall a. (Read a, Show a, Enum a, Bounded a) => Text -> Either TomlBiMapError a
```

Comment:
You don't need to write all constraints and a `forall` here. You only need to write `forall` on the top-level for `_SumTypeText` (to be renamed to `_EnumBoundedText`). Since `-XScopedTypeVariables` is already enabled for the whole project.
```suggestion
    toSumType :: Text -> Either TomlBiMapError a
```